futurex_openedx_extensions/helpers/caching.py

```diff
@@ -0,0 +1,47 @@
+"""Helper functions for caching"""
+import functools
+import logging
+
+from django.core.cache import cache
+
+log = logging.getLogger(__name__)
+
+
+def cache_dict(timeout, key_generator_or_name):
+    """Cache the dictionary result returned by the function"""
+    def decorator(func):
+        """Decorator definition"""
+        @functools.wraps(func)
+        def wrapped(*args, **kwargs):
+            """Wrapped function"""
+            cache_key = None
+            try:
+                if not isinstance(timeout, int) or timeout <= 0:
+                    raise ValueError(
+                        "unexpected timout value. Should be an integer greater than 0"
+                    )
+                if not callable(key_generator_or_name) and not isinstance(key_generator_or_name, str):
+                    raise TypeError("key_generator_or_name must be a callable or a string")
+
+                cache_key = key_generator_or_name(
+                    *args, **kwargs
+                ) if callable(key_generator_or_name) else key_generator_or_name
+
+            except Exception as exc:  # pylint: disable=broad-except
+                log.exception("cache_dict: error generating cache key: %s", exc)
```

Comment:
Since the code allows `key_generator_or_name` to be `callable` or `str`; it can fail because of a silly key generation mistake. I've tested my generators, but I mean if a new one is used without proper testing; I don't want it to block the execution of the APIs. An error message that will popup many times in the logs will be enough to the attention

I prefer keeping it passing without interrupting the API execution. This means we'll lose only the cache; not the entire response

What do you think @iamjazzar ? 
I would also appreciate your feedback @OmarIthawi

Reply:
Makes sense @shadinaif